### ISSUE TITLE
Option to use alternative CNI plugin

### ISSUE THREAD

**Issue by dangmai** [kube-hetzner/terraform-hcloud-kube-hetzner#137]
Hi,

This is a super cool project, and I really want to try it out. However, currently there's no way to disable `flannel` or network policy, which is needed to install alternative CNI plugin (like Calico) that provides some functionality I can't achieve with `flannel`. Is this something on your radar? 

Thanks!

**Comment by mysticaltech** [comment]
Hey @dangmai, basically it's pretty simple to add. 

Here are the two k3s server config flags you need to add to the k3s server config:
```
      disable-network-policy = true
      flannel-backend  = "none"
```

To be added to `init.tf` and `control_planes.tf.` And also add Calico, or Cilium for that matter, to the kustomization in `init.tf`. 

Here's a clearer video explanation: https://www.loom.com/share/393a4ad632c043ffae0d59349355adae

**Comment by mysticaltech** [comment]
PR very much welcome! :-)

**Comment by dangmai** [comment]
Yeah I do know that it can be done with k3s like you mentioned, I was just curious how you want to handle it effectively in this library.

Adding defaulted options like `flannel_enabled = true` and `network_policy_enabled = true` to `terraform.tfvars.example` is straightforward, but k3s does have quite a few knobs that are not customizable right now in this library. Maybe something like `additional_k3s_server_args = []` would be more universal?

**Comment by mysticaltech** [comment]
Definitely, both features would be welcome! 🙏

**Comment by mysticaltech** [comment]
@dangmai As cni options, valid "flannel", "calico", or "cilium". 

So "flannel", of course will be the default.
